packages/labs/react/src/create-component.ts

```diff
@@ -132,31 +170,39 @@ export const createComponent = <I extends HTMLElement, E extends Events = {}>(
   const createElement = React.createElement;
   const eventProps = new Set(Object.keys(events ?? {}));
 
-  // Props the user is allowed to use, includes standard attributes, children,
-  // ref, as well as special event and element properties.
-  type ReactProps = Omit<React.HTMLAttributes<I>, keyof E>;
-  type ElementWithoutPropsOrEvents = Omit<I, keyof E | keyof ReactProps>;
-  type UserProps = Partial<
-    ReactProps & ElementWithoutPropsOrEvents & EventProps<E>
-  >;
+  type Props = ReactComponentProps<I, E>;
 
-  // Props used by this component wrapper. This is the UserProps and the
-  // special `__forwardedRef` property. Note, this ref is special because
-  // it's both needed in this component to get access to the rendered element
-  // and must fulfill any ref passed by the user.
-  type ComponentProps = UserProps & {
-    __forwardedRef?: React.Ref<unknown>;
-  };
+  // Set of properties/events which should be specially handled by the wrapper
+  // and not handled directly by React.
+  const elementClassProps = new Set(Object.keys(events ?? {}));
```

Comment:
How is this changed related to the PR?

Reply:
i'll revert

Reply:
Oh there's nothing to revert to. This is coming across as a change but it's the same as `main`.

Reply:
I don't see this in the original file in this diff though, so it's confusing. Looks like new code. Same with the for loop below.

Do you need to merge in `main`?

Reply:
im not trying to change this part of the code

Reply:
well this PR has been sitting in limbo and had other submissions, not surprised there's side effects here

Reply:
removed